src/compiler/scala/tools/nsc/transform/Mixin.scala

```diff
@@ -1037,4 +1053,14 @@ abstract class Mixin extends InfoTransform with ast.TreeDSL {
       finally localTyper = saved
     }
   }
+
+  private def isTraitMethodRequiringStaticImpl(dd: DefDef): Boolean = {
+    val sym = dd.symbol
+    sym != null && sym.owner.isTrait && !sym.isDeferred &&
+      (sym.isLazy || !sym.isAccessor) && !sym.isSuperAccessor && !sym.isModule && // these are not deferred, but defDef.rhs.isEmpty
```

Comment:
i'm about to simplify this, it was originally much simpler (rhs.nonempty), but then i changed it to inspect the symbol only  as we needed the same test in a spot where we don't have the tree. but that's no longer the case, now that we have an attachment for that.